test/reactionsystem_structure/reactionsystem.jl

```diff
@@ -143,66 +146,29 @@ end
 # Test by evaluating drift and diffusion terms.
 
 let
-    p = rand(rng, length(k))
-    u = rand(rng, length(k))
-    du = oderhs(u, p, 0.0)
-    G = sdenoise(u, p, 0.0)
+    u = rnd_u0(rs, rng)
+    p = rnd_ps(rs, rng)
+    du = oderhs(last.(u), last.(p), 0.0)
+    G = sdenoise(last.(u), last.(p), 0.0)
     sdesys = convert(SDESystem, rs)
     sf = SDEFunction{false}(sdesys, unknowns(rs), parameters(rs))
-    du2 = sf.f(u, p, 0.0)
+    sprob = SDEProblem(rs, u, (0.0, 0.0), p)
+    du2 = sf.f(sprob.u0, sprob.p, 0.0)
+
+    du2 = sf.f(sprob.u0, sprob.p, 0.0)
     @test norm(du - du2) < 100 * eps()
-    G2 = sf.g(u, p, 0.0)
+    G2 = sf.g(sprob.u0, sprob.p, 0.0)
     @test norm(G - G2) < 100 * eps()
 
     # Test conversion to NonlinearSystem.
     ns = convert(NonlinearSystem, rs)
+    nlprob = NonlinearProblem(rs, u, p)
     fnl = eval(generate_function(ns)[2])
     dunl = similar(du)
-    fnl(dunl, u, p)
-    @test norm(du - dunl) < 100 * eps()
-end
-
-# Tests the noise_scaling argument.
-let
-    p = rand(rng, length(k) + 1)
-    u = rand(rng, length(k))
-    t = 0.0
-    G = p[21] * sdenoise(u, p, t)
-    @variables η
-    sdesys_noise_scaling = convert(SDESystem, rs; noise_scaling = η)
-    sf = SDEFunction{false}(sdesys_noise_scaling, unknowns(rs),
-                            parameters(sdesys_noise_scaling))
-    G2 = sf.g(u, p, t)
-    @test norm(G - G2) < 100 * eps()
-end
-
-# Tests the noise_scaling vector argument.
-let
-    p = rand(rng, length(k) + 3)
-    u = rand(rng, length(k))
-    t = 0.0
-    G = vcat(fill(p[21], 8), fill(p[22], 3), fill(p[23], 9))' .* sdenoise(u, p, t)
-    @variables η[1:3]
-    sdesys_noise_scaling = convert(SDESystem, rs;
-                                   noise_scaling = vcat(fill(η[1], 8), fill(η[2], 3),
-                                                        fill(η[3], 9)))
-    sf = SDEFunction{false}(sdesys_noise_scaling, unknowns(rs),
-                            parameters(sdesys_noise_scaling))
-    G2 = sf.g(u, p, t)
-    @test norm(G - G2) < 100 * eps()
-end
-
-# Tests using previous parameter for noise scaling
-let
-    p = rand(rng, length(k))
-    u = rand(rng, length(k))
-    t = 0.0
-    G = [p p p p]' .* sdenoise(u, p, t)
-    sdesys_noise_scaling = convert(SDESystem, rs; noise_scaling = k)
-    sf = SDEFunction{false}(sdesys_noise_scaling, unknowns(rs),
-                            parameters(sdesys_noise_scaling))
-    G2 = sf.g(u, p, t)
-    @test norm(G - G2) < 100 * eps()
```

Comment:
These were moved in the noise scaling PR to the SDe simulation file (so not removed).